src/lib/fileTypeGuesser.js

```diff
@@ -0,0 +1,48 @@
+export default function fileTypeGuesser(readerResult) {
+  let isJSON = true;
+  let parsedJson;
+
+  try {
+    parsedJson = JSON.parse(new TextDecoder().decode(readerResult));
+  } catch (e) {
+    if (e.message.toLowerCase().includes('is not valid json')) {
+      isJSON = false;
+    }
+  }
+
+  if (isJSON && parsedJson?.version && parsedJson?.version.startsWith('http://ipuz.org')) {
+    return 'ipuz';
+  }
+
+  const puzMagicHeader = '4143524f535326444f574e00';
```

Comment:
A `.puz` file should have this magic header as per the docs (See: https://gist.github.com/sliminality/dab21fa834eae0a70193c7cd69c356d5#header)